src/libs/antares/benchmarking/info_collectors.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <vector>
+#include <map>
+#include <string>
+#include <mutex>
+
+#include <antares/study.h>
+#include "file_content.h"
+
+
+namespace Benchmarking
+{
+
+struct OptimizationInfo
+{
+    unsigned int nbVariables = 0;
+    unsigned int nbConstraints = 0;
+    unsigned int nbNonZeroCoeffs = 0;
+};
+
+/*
+    === Info collectors ===
+*/
+
+class StudyInfoCollector
+{
+public:
+    StudyInfoCollector(const Antares::Data::Study& study)
+        : study_(study)
+    {}
+    void toFileContent(FileContent& file_content);
+private:
+    // Methods
+    void areasCountToFileContent(FileContent& file_content);
+    void linksCountToFileContent(FileContent& file_content);
+    void performedYearsCountToFileContent(FileContent& file_content);
+    void enabledThermalClustersCountToFileContent(FileContent& file_content);
+    void enabledBindingConstraintsCountToFileContent(FileContent& file_content);
+    void unitCommitmentModeToFileContent(FileContent& file_content);
+    void maxNbYearsInParallelToFileContent(FileContent& file_content);
+    void solverVersionToFileContent(FileContent& file_content);
+
+    // Member data
+    const Antares::Data::Study& study_;
+
+    // ----------------------------------------------------------------------------------------
+    // TODO : add some more pieces of information about study : see following data members.
+    // ----------------------------------------------------------------------------------------
+    // Parallel execution
+    unsigned int minNbYearsInParallel_ = 0;
+    unsigned int nbCoreMode_ = 0;
+};
+
+class SimulationInfoCollector
+{
+public:
+    SimulationInfoCollector(const OptimizationInfo& optInfo)
+        : opt_info_(optInfo)
+    {};
+
+    void toFileContent(FileContent& file_content);
+
+private:
+    const OptimizationInfo& opt_info_;
+};
+
+
+
+class IDurationCollector
+{
+public:
+    virtual void addDuration(std::string name, int64_t duration) = 0;
+};
+
+class NullDurationCollector : public IDurationCollector
+{
+public:
+    NullDurationCollector() = default;
+    void addDuration(std::string name, int64_t duration) override { /* Do nothing */ }
+};
+
+class DurationCollector : public IDurationCollector
+{
+public:
+    DurationCollector() = default;
+
+    void toFileContent(FileContent& file_content);
+    void addDuration(std::string name, int64_t duration) override;
+
+private:
+    map<string, vector<int64_t>> duration_items_;
```

Comment:
```suggestion
    map<string, int64_t> duration_items_;
```
There is no need to keep all durations, only their sum.

Reply:
See remark above